packages/side-nav/src/vaadin-side-nav-item.js

```diff
@@ -9,13 +9,65 @@ import { DisabledMixin } from '@vaadin/a11y-base/src/disabled-mixin.js';
 import { defineCustomElement } from '@vaadin/component-base/src/define.js';
 import { ElementMixin } from '@vaadin/component-base/src/element-mixin.js';
 import { PolylitMixin } from '@vaadin/component-base/src/polylit-mixin.js';
+import { TooltipController } from '@vaadin/component-base/src/tooltip-controller.js';
 import { matchPaths } from '@vaadin/component-base/src/url-utils.js';
 import { LumoInjectionMixin } from '@vaadin/vaadin-themable-mixin/lumo-injection-mixin.js';
 import { ThemableMixin } from '@vaadin/vaadin-themable-mixin/vaadin-themable-mixin.js';
 import { location } from './location.js';
 import { sideNavItemStyles } from './styles/vaadin-side-nav-item-base-styles.js';
 import { SideNavChildrenMixin } from './vaadin-side-nav-children-mixin.js';
 
+/**
+ * A controller for observing tooltip slotted in the item.
+ * @private
+ */
+export class ItemTooltipController extends TooltipController {
+  constructor(host) {
+    super(host);
+    this.setTarget(host.$.content);
+    this.setAriaTarget(null);
+  }
+
+  /**
+   * @protected
+   * @override
+   */
+  initCustomNode(node) {
+    super.initCustomNode(node);
+    this.observeNode(node);
+  }
+
+  /**
+   * Setup the mutation observer on the node to update ID and notify host.
+   * Node doesn't get observed automatically until this method is called.
+   *
+   * @param {Node} node
+   * @private
+   */
+  observeNode(node) {
+    // Stop observing the previous node, if any.
+    if (this.__nodeObserver) {
+      this.__nodeObserver.disconnect();
+    }
+
+    this.__nodeObserver = new MutationObserver((mutations) => {
```

Comment:
This approach seems quite heavy. Maybe we could just fire an internal event in the tooltip handler that updates the content?

Reply:
Created a PR: https://github.com/vaadin/web-components/pull/10045